core/src/main/scala/com/pingcap/tispark/TiBatchWrite.scala

```diff
@@ -710,9 +711,51 @@ class TiBatchWrite(@transient val df: DataFrame,
     tiTableInfo.getId
   }
 
+  private def estimateRegionSplitNum(wrappedRowRdd: RDD[WrappedRow]) = {
+    //TODO refine this https://github.com/pingcap/tispark/issues/891
+    val rowSize = tiTableInfo.getEstimatedRowSizeInByte
+    //TODO: replace 96 with actual value read from pd https://github.com/pingcap/tispark/issues/890
+    (wrappedRowRdd.count() * rowSize) / (96 * 1024 * 1024)
+  }
+
+  private def splitIndexRegion(wrappedRowRdd: RDD[WrappedRow]) = {
+    if (options.enableRegionSplit && isEnableSplitRegion) {
+      val regionSplitNum = if (options.regionSplitNum != 0) {
+        options.regionSplitNum
+      } else {
+        estimateRegionSplitNum(wrappedRowRdd)
```

Comment:
current implementation of `estimateRegionSplitNum` is only suitable for `splitTableRegion`, better to implement a special version for `splitIndexRegion`

because `index data size` is usually much smaller than `table data size`

Reply:
I had the same concern, let me figure out a way to resolve this.

Reply:
resolved.